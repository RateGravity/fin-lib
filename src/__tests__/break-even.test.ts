import { computeBreakEven } from '@ownup/fin-lib';
import given from 'given2';
import { cases } from './break-even-cases';

describe('computeBreakEven', () => {
  given('breakEven', () => computeBreakEven(given.inputA, given.inputB));

  // Breakeven is hard math. We've validated the cases that
  // are coded here by manually running the amortizations tables
  // and finding the breakeven point, using snapshots to effectively
  // write these tests and prevent them from changing is easier than
  // pretending we have well understood answers.
  it.each(cases)('given %j', ({ a, b }) => {
    given('inputA', () => a);
    given('inputB', () => b);
    expect(given.breakEven).toMatchSnapshot();
  });
});
