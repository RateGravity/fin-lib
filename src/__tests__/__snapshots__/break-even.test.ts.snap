// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`computeBreakEven given {"a":{"totalFees":0,"initialRate":1,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0},"b":{"totalFees":100000,"initialRate":10,"presentValue":500000,"loanTerm":15,"propertyValue":600000,"mortgageInsurance":0}} 1`] = `-Infinity`;

exports[`computeBreakEven given {"a":{"totalFees":0,"initialRate":3.75,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0},"b":{"totalFees":1000,"initialRate":4.75,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0}} 1`] = `-Infinity`;

exports[`computeBreakEven given {"a":{"totalFees":0,"initialRate":3.75,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0},"b":{"totalFees":10000,"initialRate":3,"presentValue":500000,"loanTerm":15,"propertyValue":600000,"mortgageInsurance":0}} 1`] = `-6`;

exports[`computeBreakEven given {"a":{"totalFees":0,"initialRate":3.75,"presentValue":600000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0},"b":{"totalFees":0,"initialRate":3.75,"presentValue":600000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0}} 1`] = `0`;

exports[`computeBreakEven given {"a":{"totalFees":0,"initialRate":5,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0},"b":{"totalFees":10000,"initialRate":3,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0}} 1`] = `-10`;

exports[`computeBreakEven given {"a":{"totalFees":1000,"initialRate":3,"presentValue":500000,"loanTerm":15,"propertyValue":600000,"mortgageInsurance":0},"b":{"totalFees":5000,"initialRate":1,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0}} 1`] = `-336`;

exports[`computeBreakEven given {"a":{"totalFees":1000,"initialRate":4.75,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0},"b":{"totalFees":0,"initialRate":3.75,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0}} 1`] = `Infinity`;

exports[`computeBreakEven given {"a":{"totalFees":5000,"initialRate":1,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0},"b":{"totalFees":1000,"initialRate":3,"presentValue":500000,"loanTerm":15,"propertyValue":600000,"mortgageInsurance":0}} 1`] = `336`;

exports[`computeBreakEven given {"a":{"totalFees":10000,"initialRate":3,"presentValue":500000,"loanTerm":15,"propertyValue":600000,"mortgageInsurance":0},"b":{"totalFees":0,"initialRate":3.75,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0}} 1`] = `6`;

exports[`computeBreakEven given {"a":{"totalFees":10000,"initialRate":3,"presentValue":500000,"loanTerm":15,"propertyValue":600000,"mortgageInsurance":0},"b":{"totalFees":100000,"initialRate":3.75,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0}} 1`] = `-Infinity`;

exports[`computeBreakEven given {"a":{"totalFees":10000,"initialRate":3,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0},"b":{"totalFees":0,"initialRate":5,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0}} 1`] = `10`;

exports[`computeBreakEven given {"a":{"totalFees":100000,"initialRate":3.75,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0},"b":{"totalFees":10000,"initialRate":3,"presentValue":500000,"loanTerm":15,"propertyValue":600000,"mortgageInsurance":0}} 1`] = `Infinity`;

exports[`computeBreakEven given {"a":{"totalFees":100000,"initialRate":10,"presentValue":500000,"loanTerm":15,"propertyValue":600000,"mortgageInsurance":0},"b":{"totalFees":0,"initialRate":1,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0}} 1`] = `Infinity`;

exports[`computeBreakEven given {"a":{"totalFees":10000000,"initialRate":3,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0},"b":{"totalFees":0,"initialRate":5,"presentValue":500000,"loanTerm":30,"propertyValue":600000,"mortgageInsurance":0}} 1`] = `Infinity`;
